src/Tasks/XslTransformation.cs

```diff
@@ -334,7 +335,7 @@ public XmlReader CreateReader(int itemPos)
             {
                 if (XmlMode == XmlModes.XmlFile)
                 {
-                    return XmlReader.Create(_data[itemPos]);
+                    return XmlReader.Create(new StreamReader(_data[itemPos], new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), detectEncodingFromByteOrderMarks: true));
```

Comment:
Have you validated that these encoding assumptions are valid for arbitrary user XML (as opposed to project files)?

Reply:
I haven't. I've been poking around with this fix quite a bit now, maybe you'll have a better idea.

This fix is a bit overkill, all that's really needed is `new StreamReader(_data[itemPos])`. But there's a single failure in `XslDocumentFunctionWorks` that is somehow looking for `b.xml` in the bin directory of the running unit test that's very confusing. It occurs right as it's attempting to transform an xslt file